### PR TITLE
Update Keycloak to Quarkus 2.13.8.Final

### LINKED ISSUE #20562: Update Keycloak to Quarkus 2.13.8.Final
### Description

The Quarkus team is planning the 2.13.8.Final release, as soon as the release is out we must update the `release/21.1` branch, which contains significant updates and improvements. As soon as the release is out, we must update to benefit from relevant CVE fixes.

### PR DESCRIPTION
**Pull request by abstractj**
Closes #20562

### CODE REVIEW

**Review by vmuzikar — APPROVED** [review]
LGTM. The test failures are unrelated, I believe. Re-triggered the CI just to be sure.

**Comment by abstractj** [comment]
@keycloak/store-maintainers, could you please review the changes proposed here? I noticed some failures on the Store Model tests, but I'm not sure if they are related.

**Review by hmlnarik — CHANGES_REQUESTED** [review]
The test failures seem related, and for that reason I'm now marking this PR as changes requested. I'm looking into the failed tests now.

File: pom.xml [hunk 1]
```diff
@@ -79,10 +79,10 @@
         <h2.version>2.1.214</h2.version>
         <jakarta.persistence.version>2.2.3</jakarta.persistence.version>
         <jakarta.servlet.jakarta-servlet-api.version>6.0.0</jakarta.servlet.jakarta-servlet-api.version>
-        <hibernate-orm.version>5.6.14.Final</hibernate-orm.version>
+        <hibernate-orm.version>5.6.15.Final</hibernate-orm.version>
         <hibernate.c3p0.version>${hibernate-orm.version}</hibernate.c3p0.version>
-        <infinispan.version>14.0.7.Final</infinispan.version>
-        <infinispan.protostream.processor.version>4.6.0.Final</infinispan.protostream.processor.version>
+        <infinispan.version>14.0.9.Final</infinispan.version>
```
Comment (hmlnarik) [inline pom.xml:84]: This line causes the failure.

**Comment by hmlnarik** [comment]
The errors are caused by the following missing commit: #20069

Feel free to either backport it yourself or take over https://github.com/hmlnarik/keycloak/tree/20562-Update-Keycloak-to-Quarkus-2-13-8-Final-main where the backport including conflict resolution for this branch is prepared. I have inserted the 14.0.8 commit before the Quarkus upgrade commit to keep the time order correct.

File: pom.xml (hunk 1 above)
Comment (abstractj) [reply pom.xml:84]: @hmlnarik thanks for looking into this. Do you think is ok to move to 14.0.9 if we perform the backport?

Comment (hmlnarik) [reply pom.xml:84]: Yup. The https://github.com/hmlnarik/keycloak/tree/20562-Update-Keycloak-to-Quarkus-2-13-8-Final-main has 14.0.9.

**Comment by abstractj** [comment]
@hmlnarik all green now, thanks for pointing me in the right direction!

**Review by vmuzikar — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**